examples/training/semantic_segmentation/pascal_voc/deeplab_v3.py

```diff
@@ -56,49 +55,80 @@
+# all_ds = all_ds.concatenate(load(split="sbd_eval", data_dir=None))
+# train_ds = all_ds.take(10000)
+# eval_ds = all_ds.skip(10000).concatenate(load(split="diff", data_dir=None))
 train_ds = load(split="sbd_train", data_dir=None)
 eval_ds = load(split="sbd_eval", data_dir=None)
 
-resize_layer = tf.keras.layers.Resizing(512, 512)
+resize_layer = tf.keras.layers.Resizing(512, 512, interpolation="nearest")
+
+image_size = [512, 512, 3]
+
+
+def flip_fn(image, cls_seg):
+    if tf.random.uniform([], minval=0, maxval=1, dtype=tf.float32) > 0.5:
+        image = tf.image.flip_left_right(image)
+        cls_seg = tf.reverse(cls_seg, axis=[1])
+    return image, cls_seg
 
 
 def proc_train_fn(examples):
     image = examples.pop("image")
     image = tf.cast(image, tf.float32)
     image = resize_layer(image)
     cls_seg = examples.pop("class_segmentation")
+    cls_seg = tf.cast(cls_seg, tf.float32)
+    cls_seg = resize_layer(cls_seg)
+    image, cls_seg = flip_fn(image, cls_seg)
+    cls_seg = tf.cast(cls_seg, tf.uint8)
+    sample_weight = tf.equal(cls_seg, 255)
     zeros = tf.zeros_like(cls_seg)
-    mask = tf.equal(cls_seg, 255)
-    cls_seg = tf.where(mask, zeros, cls_seg)
+    cls_seg = tf.where(sample_weight, zeros, cls_seg)
+    return image, cls_seg
+
+
+def proc_eval_fn(examples):
+    image = examples.pop("image")
+    image = tf.cast(image, tf.float32)
+    image = resize_layer(image)
+    cls_seg = examples.pop("class_segmentation")
+    cls_seg = tf.cast(cls_seg, tf.float32)
     cls_seg = resize_layer(cls_seg)
-    cls_seg = tf.cast(cls_seg, tf.uint32)
+    cls_seg = tf.cast(cls_seg, tf.uint8)
+    sample_weight = tf.equal(cls_seg, 255)
+    zeros = tf.zeros_like(cls_seg)
+    cls_seg = tf.where(sample_weight, zeros, cls_seg)
     return image, cls_seg
 
 
 train_ds = train_ds.map(proc_train_fn, num_parallel_calls=tf.data.AUTOTUNE)
 train_ds = train_ds.batch(global_batch)
 
-eval_ds = eval_ds.map(proc_train_fn, num_parallel_calls=tf.data.AUTOTUNE)
+eval_ds = eval_ds.map(proc_eval_fn, num_parallel_calls=tf.data.AUTOTUNE)
 eval_ds = eval_ds.batch(global_batch)
 
 train_ds = train_ds.shuffle(8)
+train_ds = train_ds.prefetch(2)
 
 with strategy.scope():
     lr_decay = tf.keras.optimizers.schedules.PiecewiseConstantDecay(
         boundaries=[30000 * 16 / global_batch],
         values=[base_lr, 0.1 * base_lr],
     )
-    backbone = keras_cv.models.ResNet50V2(
-        include_rescaling=True, weights="imagenet", include_top=False
-    ).as_backbone()
-    model = DeepLabV3(classes=21, backbone=backbone, include_rescaling=True)
+    model = DeepLabV3(
+        classes=21, backbone="resnet50_v2", include_rescaling=True, weights="imagenet"
+    )
     optimizer = tf.keras.optimizers.SGD(
         learning_rate=lr_decay, momentum=0.9, clipnorm=10.0
     )
-    loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)
+    loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(ignore_class=255)
```

Comment:
Since 255 is a magic number in this dataset, should we export it as a constant from the dataloader with a descriptive name?

Reply:
We paid for something like this from other keras datasets (such as mnist) when we attach class_ids to it. But the issue is that this will get lost when adding another transformation on the returned dataset.
So adding a comment here for users who need to copy the template.

Reply:
Done.